### ISSUE TITLE
Adopting existing nodes for karpenter migration

### ISSUE THREAD

**Issue by ilkinmammadzada** [aws/karpenter#4176]
Hey everyone!
I want to start with karpenter, but don't want to delete existing nodes/EC2s.

I have checked EC2 tags and nodes labels/annotations and added all of them to the existing nodes/EC2s. Also I have created machine CR for all my nodes. But when I start karpenter it detects them as "garbage" and deletes all of them:
`DEBUG	controller.machine.garbagecollection	garbage collected cloudprovider machine	{"commit": "26e2d35-dirty", "provider-id": "aws:///us-west-1c/i-xxxx"}`

I cannot create machine object properly, I think this may be reason. I mean I cannot add status section to the object, namely nodeName, providerID. they are deleted automatically, also cannot change those values for existing machine object (which created by karpenter) 

Anyone has idea for migrating to karpenter with adopting existing nodes?

**Comment by jonathan-innis** [comment]
How are you adopting existing Machines over to Karpenter? Are you adding tags to existing instances? Adding labels to the nodes?

**Comment by ilkinmammadzada** [comment]
> How are you adopting existing Machines over to Karpenter? Are you adding tags to existing instances? Adding labels to the nodes?

Yes, exactly. Also adding an annotation and finalizer to the node objects. And sure creating machine object. But actually it didn't help, karpenter GC removed all EC2s. I think because we couldn't add ec2 id and node name to the machine object (those fields live in .status section)

**Comment by jonathan-innis** [comment]
> Also adding an annotation and finalizer to the node objects

We don't generally support the adoption of instances owned by MNG or just instances that joined the cluster as standalone instances. We generally recommend that you scale-down these existing instances slowly and let Karpenter scale-up new instances that are owned by the Provisioner and have the proper tags/labels for the instance that was terminated. I'd recommend this method over trying to adopt these existing instances into Karpenter.

**Comment by ilkinmammadzada** [comment]
@jonathan-innis  Thanks for recommendation! But it's really headache if you have many tenants/nodes/clusters :) That's why I wanted to migrate without scaling-down/up.

**Comment by jonathan-innis** [comment]
> That's why I wanted to migrate without scaling-down/up

Yeah, you're just really in unknown territory if you go this route. I'm surprised it's more of a headache to go this route than to have to tag all the instances and identify a Provisioner that they should all be associated with.

Are any of these instance owned by other components? If the instances were managed by MNG, it seems like tagging them as Karpenter nodes might create some weird behavior, particularly when you scale down or delete the MNG.
